src/transformers/generation_utils.py

```diff
@@ -792,6 +806,12 @@ def generate(
                 crash. Note that using ``remove_invalid_values`` can slow down generation.
             synced_gpus (:obj:`bool`, `optional`, defaults to :obj:`False`):
                 Whether to continue running the while loop until max_length (needed for ZeRO stage 3)
+            logits_processor (:obj:`LogitsProcessorList`, `optional`):
+                This object is created automatically from other arguments of this function. `logits_processor` is meant
```

Comment:
Not a huge fan of the docstring here - could we maybe rephrase it a bit. I think the user is mostly interested in what happens when this object is passed and how it can be passed - not really what happens when it is not passed.

So maybe something more like:
```suggestion
                If provided `logits_processor` will overwrite all passed arguments that can process logits as well as those saved in the model's config. It can be very useful to enable custom logits processing logic. 
```

We should also note somewhere that this is an experimental feature IMO

Reply:
keen to hear your input here as well @Narsil . IMO this feature is really not for the "unexperienced" HF user but for the advanced ones that know more or less what happens under the hood in `generate()` (otherwise why need costum logit processors?). To give this functionality while keeping complexity at a minimum I think the best first step is to simply say: 
If we pass `logits_processor` or `stopping_criteria` it will overwrite everything else...

Reply:
That's what I named `Option 2` above. While I think it's viable I don't think it's the only way.
`Option 1` which is ("we're adding your stuff too without looking") is also perfectly viable. Let's continue discussion above since I think that's where the main point is, no ?